distributed/comm/tcp.py

```diff
@@ -387,7 +387,13 @@ async def connect(self, address, deserialize=True, **connection_args):
             # The socket connect() call failed
             convert_stream_closed_error(self, e)
         except SSLError as err:
-            raise FatalCommClosedError() from err
+            if "CERTIFICATE_VERIFY_FAILED" in str(err):
+                raise FatalCommClosedError(
+                    "TLS certificate does not match. Check your security settings. "
+                    "More info at https://distributed.dask.org/en/latest/tls.html"
+                ) from err
+            else:
+                raise FatalCommClosedError() from err
```

Comment:
Instead of checking the error message that's raised for `CERTIFICATE_VERIFY_FAILED`, can we instead add an `except SSLCertVerificationError` branch before the current `except SSLError` branch. Something along the lines of (the snippet below is totally untested):

```python
try:
    ...
except StreamClosedError as e:
    ...
except SSLCertVerificationError as err:
    raise FatalCommClosedError(
        "TLS certificate does not match. Check your security settings. "
        "More info at https://distributed.dask.org/en/latest/tls.html"
    ) from err
except SSLError as err:
    raise FatalCommClosedError() from err
```

This should be equivalent to what you currently have, but slightly more robust to changes in the error message which is raised

Reply:
Good plan thanks @jrbourbeau